### PR TITLE
feat: storage.lua

### PR DESCRIPTION
**Pull request by mafewtm**
## Description

Implement a storage.lua

## Checklist

- [ ] I have personally loaded this code into an updated Qbox project and checked all of its functionality.
- [ ] My code fits the style guidelines.
- [ ] My PR fits the contribution guidelines.

### CODE REVIEW

File: server/storage.lua [hunk 1]
```diff
@@ -0,0 +1,20 @@
+---@param groupName string
+---@param group string
```
Comment (Manason) [inline server/storage.lua]: what is the difference between groupName and group?

File: server/storage.lua [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+---@return Player[]
+function FetchPlayerByCitizenId(citizenId)
+    return MySQL.query.await('SELECT * FROM players WHERE citizenid = ? LIMIT 1', {citizenId})
+end
+
+---@param column string
+---@param role string
+---@param citizenId string
+---@return Player[]
+function UpdatePlayerJob(column, role, citizenId)
```
Comment (Manason) [inline server/storage.lua]: Feels like citizenid should go first as the identifier?

File: server/storage.lua [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+end
+
+---@param citizenId string
+---@return Player[]
+function FetchPlayerByCitizenId(citizenId)
+    return MySQL.query.await('SELECT * FROM players WHERE citizenid = ? LIMIT 1', {citizenId})
+end
+
+---@param column string
+---@param role string
```
Comment (Manason) [inline server/storage.lua]: Maybe role should be passed as a table and then json encoded in this function? Feels like that should happen as late as possible

File: server/storage.lua [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+    return MySQL.query.await("SELECT * FROM `players` WHERE ?? LIKE '%"..groupName.."%'", {group})
+end
+
+---@param citizenId string
+---@return Player[]
+function FetchPlayerByCitizenId(citizenId)
+    return MySQL.query.await('SELECT * FROM players WHERE citizenid = ? LIMIT 1', {citizenId})
+end
+
+---@param column string
```
Comment (Manason) [inline server/storage.lua]:
```suggestion
---@param column gang | job
```

File: server/storage.lua [hunk 5]
```diff
@@ -0,0 +1,20 @@
+---@param groupName string
+---@param group string
+---@return Players[]
+function FetchPlayersByGroup(groupName, group)
```
Comment (Manason) [inline server/storage.lua]:
```suggestion
function FetchPlayerEntitiesByGroup(groupName, group)
```

File: server/storage.lua [hunk 6]
```diff
@@ -0,0 +1,20 @@
...
+    return MySQL.query.await("SELECT * FROM `players` WHERE ?? LIKE '%"..groupName.."%'", {group})
+end
+
+---@param citizenId string
+---@return Player[]
+function FetchPlayerByCitizenId(citizenId)
```
Comment (Manason) [inline server/storage.lua]:
```suggestion
function FetchPlayerEntityByCitizenId(citizenId)
```

**Review by Manason — CHANGES_REQUESTED** [review]
Feel like we should refer to the database table returned as an Entity to differentiate from the player objects we use in core

File: server/storage.lua (hunk 1 above)
Comment (mafewtm) [reply server/storage.lua]: image

**Review by Manason — APPROVED** [review]

**Pull request merged**